Add summary dashboard for PerfKitBenchmarker.

yuyantingzero opened the pull request.

cmccoy commented:
* Please remove the mongodb benchmark.
* (at least) the netperf link does not work for me.
* Consider adding some more netperf metrics.

jmuharsky commented:
The SQL statements provided in the json dashboard are still using the {{dataset_name}} and {{table_name}} placeholders.

yuyantingzero commented:
@cmccoy  :
- Removed mongodb benchmark.
- The link points to dive-in dashboards, which are not ready in this PR. I will fix the links later in next PRs.
- Added TCP_RR for netperf.
I will leave the dashboard file for now, only updating the dashboards on the experiment Explorer site. And download and update the json file later in this PR.
@jmuharsky :
The SQL statement in this PR uses {{dataset_name}}, {{table_name}}. In the README.md file, there are instructions to replace dataset_name, table_name with target dataset and table.
I can use samples_mart, results directly if you think that is enough for most users. I will send you the deployed preview dashboards offline.

cmccoy commented:
@yuyantingzero: Thanks.
What is the criterion for including a result from a benchmark in this summary page?

yuyantingzero commented:
@cmccoy I am trying to include 1 or 2 most important metric per benchmark.

cmccoy commented:
* Could you note that this uses the default machine type in the header for the first container?
* How are new clouds going to be handled? Could you use pivots?
* There are two "netperf" rows - maybe just have one label for the netperf group?

Otherwise, LGTM.

jmuharsky commented:
@yuyantingzero We shouldn't be varying from the samples_mart.results pattern, so I think fixing these is the right thing to do.  We can instruct the user to change those values if they should upload to a different dataset/table, but our tooling should guide them towards these values.

jmuharsky commented:
Recommend a more descriptive title than _PerfkitBenchmarker -- something like PerfKit Benchmarker Summary Dashboard (Sample).

yuyantingzero commented:
@jmuharsky Addressed all comments.
@cmccoy Used pivots in summary dashboard now, but since query has a different set of results, the color schema is different in each widget, any suggestions?

cmccoy commented:
LGTM modulo missing data.

jmuharsky commented on data/dashboards/README.md:
```diff
@@ -0,0 +1,19 @@
+
+* Configure the the dashboards with your BIGQUERY_PROJECT_ID used by PerfKitExplorer by running the following command in this directory:
+
+        sed -i  's/{{ project_id }}/BIGQUERY_PROJECT_ID/g' *.json
```
The dashboard included in this PR seems to rely on the default project from the Explorer config.  I'd recommend setting the project_id attribute of the dashboard to BIGQUERY_PROJECT_ID so it gets picked up by this statement.  Otherwise, remove this statement, and document that the user can set the default project for the dashboard on the Dashboard tab of Explorer after opening it.

jmuharsky commented:
Recommend moving the legends to top, bottom or inline/right, and providing enough "left" on the chart area to account for the x axis.

yuyantingzero commented:
Addressed all comments, PTAL @jmuharsky

yuyantingzero commented:
@jmuharsky Updated color and ordering.

jmuharsky commented:
LGTM
